java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/HistoricalLog.java

```diff
@@ -122,13 +124,16 @@ public synchronized void buildHistory(
         .append('\n');
 
     if (firstEvent != null) {
+      long time = firstEvent.time;
+      String note = firstEvent.note;
+      final StackTrace stackTrace = firstEvent.stackTrace;
       sb.append(innerIndentation)
-          .append(firstEvent.time)
+          .append(time)
           .append(' ')
-          .append(firstEvent.note)
+          .append(note)
           .append('\n');
       if (includeStackTrace) {
-        firstEvent.stackTrace.writeToBuilder(sb, indent + 2);
+        stackTrace.writeToBuilder(sb, indent + 2);
```

Comment:
nit: why introducing new variables?

Reply:
This is just a refactor of two identical lines of code to output the stack trace to a StringBuilder.